cpp/src/ray/test/cluster/cluster_mode_test.cc

```diff
@@ -228,6 +228,53 @@ TEST(RayClusterModeTest, ExceptionTest) {
   EXPECT_THROW(object1.Get(), ray::internal::RayTaskException);
 }
 
+bool CheckRefCount(const std::string &object_id,
+                   std::pair<size_t, size_t> ref_count_pair) {
+  auto map = ray::internal::GetRayRuntime()->GetAllReferenceCounts();
+  auto it = map.find(object_id);
+  if (it == map.end()) {
+    return false;
+  }
+
+  return it->second == ref_count_pair;
+}
+
+bool CheckRefCount(const std::string &object_id) {
+  auto map = ray::internal::GetRayRuntime()->GetAllReferenceCounts();
+  return map.find(object_id) == map.end();
+}
+
+TEST(RayClusterModeTest, LocalRefrenceTest) {
+  std::string objec_id;
+  {
+    auto r1 = ray::Task(Return1).Remote();
+    objec_id = r1.ID();
+    EXPECT_TRUE(CheckRefCount(objec_id, std::make_pair(1, 0)));
+
+    ray::ObjectRef<int> r2(r1);
+    EXPECT_TRUE(CheckRefCount(objec_id, std::make_pair(2, 0)));
+    // r1.Get();
+    // r2.Get();
+  }
+  EXPECT_TRUE(CheckRefCount(objec_id));
+}
+
+TEST(RayClusterModeTest, DependencyRefrenceTest) {
```

Comment:
What test case in Python or Java does this test cover?

Reply:
https://github.com/ray-project/ray/blob/master/java/test/src/main/java/io/ray/test/ReferenceCountingTest.java#L168